addOns/pscanrulesAlpha/src/main/zapHomeFiles/xml/js-function-list.txt

```diff
@@ -0,0 +1,9 @@
+# This file lists all dangerous JS functions being scanned for
+bypassSecurityTrustHtml
+bypassSecurityTrustScript
+bypassSecurityTrustStyle
+bypassSecurityTrustUrl
+bypassSecurityTrustResourceUrl
+trustAsHtml
+$eval
+$evalAsync
```

Comment:
This seems limited to Angular functions.

It could include `eval()` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval) and likely functions from other major frameworks as well.

Reply:
I added `eval` but I'm not too familiar with JS so I don't really know what dangerous functions from other frameworks are. I tried a little googling but only came across `document.write()` and even then I'm not too sure if it's dangerous in the same sense as these others (e.g. for XSS). 

It's why I put the list as a file so it can easily be added to later on without any need to understand the scanner. If you have any good places for me to learn more I'm very willing :)

Reply:
Do you feel like adding support for Custom Payloads? Then users can add things right within ZAP and not have to even edit files. (If you're up for it, I can find an example PR you can follow.)

Reply:
I could give it a shot yea, if I get lost I'll try asking around somewhere 👍

Reply:
Here's a few you can refer to:
- https://github.com/zaproxy/zap-extensions/pull/2287/files
- https://github.com/zaproxy/zap-extensions/pull/2215/files
- https://github.com/zaproxy/zap-extensions/pull/2213/files

Definitely let us know if you have any trouble.

Reply:
> Definitely let us know if you have any trouble.

Just curious actually, do you guys have any preferred channel for help seekers? Between the slack, IRC, and google dev group I don't really know which channel is appropriate haha

Reply:
If you want "live" assistance your best bet is to try IRC. 

https://zaproxy.org/irc/

If you have questions on use or development that might need more group consensus etc then the Dev group.

(For my 2 cents anyway, I think the others will agree though.)

Reply:
I took a look at the PRs and am wondering whether to create an `ExtensionPayloader` in  `pscanAlpha` or import as a dependency the one from `pscanrules`? If I do the latter however I will instead then need to import into `pscanrules` my extension which would require an additional dependency import regardless, so I'm guessing I should go with the former?

Reply:
Yes, better to create the `ExtensionPayloader`.